File: src/Core/WMS.Application/Features/Dock/Appointments/Dtos/DockAppointmentDto.cs
using WMS.Domain.Enums;

public record DockAppointmentDto
{
    public Guid Id { get; init; }
    public Guid DockId { get; init; }
    public string DockName { get; init; } = string.Empty;  
    public Guid TruckId { get; init; }
    public string LicensePlate { get; init; } = string.Empty;
    public DateTime StartDateTime { get; init; }
    public DateTime EndDateTime { get; init; }
    public AppointmentStatus Status { get; init; }
    public string? YardSpotNumber { get; init; }  
}

File: wms-frontend/src/app/core/models/dock-appointment.dto.ts

export interface DockAppointmentDto {
  id: string;
  dockId: string;
  dockName: string;
  truckId: string;
  licensePlate: string;
  startDateTime: string;
  endDateTime: string;
  status: 'Scheduled' | 'InProgress' | 'Unloading' | 'Completed' | 'Cancelled';
  yardSpotNumber?: string;
}
